From 9458cc6b5aab41edd14bb7a5b0e93484b46bb142 Mon Sep 17 00:00:00 2001
From: Antti Ajanki <antti.ajanki@iki.fi>
Date: Mon, 29 Apr 2013 22:22:22 +0300
Subject: [PATCH 019/110] [graphics] Fix "can't use default assignment
 operator"

by dropping the TextData reference, it's not being used, anyway
---
 src/backends/graphics.cpp | 3 +--
 src/backends/graphics.h   | 6 ++----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/backends/graphics.cpp b/src/backends/graphics.cpp
index a395eaf..f6ac15c 100644
--- a/src/backends/graphics.cpp
+++ b/src/backends/graphics.cpp
@@ -973,8 +973,7 @@ std::vector<LineData> CairoPangoRenderer::getLineData(const TextData& _textData)
 		PangoRectangle rect;
 		pango_layout_iter_get_line_extents(lineIter, NULL, &rect);
 		PangoLayoutLine* line = pango_layout_iter_get_line(lineIter);
-		data.emplace_back(_textData,
-				  PANGO_PIXELS(rect.x) - XOffset,
+		data.emplace_back(PANGO_PIXELS(rect.x) - XOffset,
 				  PANGO_PIXELS(rect.y) - YOffset,
 				  PANGO_PIXELS(rect.width),
 				  PANGO_PIXELS(rect.height),
diff --git a/src/backends/graphics.h b/src/backends/graphics.h
index 4ffa53e..674a967 100644
--- a/src/backends/graphics.h
+++ b/src/backends/graphics.h
@@ -369,16 +369,14 @@ public:
 
 class LineData {
 public:
-	LineData(const TextData& o, int32_t x, int32_t y, int32_t _width,
+	LineData(int32_t x, int32_t y, int32_t _width,
 		 int32_t _height, int32_t _firstCharOffset, int32_t _length,
 		 number_t _ascent, number_t _descent, number_t _leading,
 		 number_t _indent):
-		owner(o), extents(x, x+_width, y, y+_height), 
+		extents(x, x+_width, y, y+_height), 
 		firstCharOffset(_firstCharOffset), length(_length),
 		ascent(_ascent), descent(_descent), leading(_leading),
 		indent(_indent) {}
-	// This line belongs to owner
-	const TextData& owner;
 	// position and size
 	RECT extents;
 	// Offset of the first character on this line
-- 
1.8.5.3

